gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -2364,14 +2373,58 @@ else if (value instanceof Traversal)
-                this.property((VertexProperty.Cardinality) key, value, null == keyValues ? null : keyValues[0],
+        if (key instanceof VertexProperty.Cardinality) {
+            if (value instanceof Map) { //Handle the property(Cardinality, Map) signature
+                final Map<Object, Object> map = (Map)value;
+                for (Map.Entry<Object, Object> entry : map.entrySet()) {
+                    property(key, entry.getKey(), entry.getValue());
+                }
+                return this;
+            } else {
+                return this.property((VertexProperty.Cardinality) key, value, null == keyValues ? null : keyValues[0],
                         keyValues != null && keyValues.length > 1 ?
                                 Arrays.copyOfRange(keyValues, 1, keyValues.length) :
-                                new Object[]{}) :
-                this.property(null, key, value, keyValues);
+                                new Object[]{});
+            }
+        } else  { //handles if cardinality is not the first parameter
+            if (value instanceof Map) { //Handle the property(Cardinality, Map) signature
+                final Map<Object, Object> map = (Map)value;
+                for (Map.Entry<Object, Object> entry : map.entrySet()) {
+                    property(null, entry.getKey(), entry.getValue());
+                }
+                return this;
+            } else {
+                return this.property(null, key, value, keyValues);
+            }
+        }
+    }
+    
+    /**
+     * When a {@link Map} is supplied and if supported by the {@link Graph} then each of the key/value pairs in the map will
+     * be added as property.  This method is the long-hand version of looping through the 
+     * {@link #property(Object, Object, Object...)} method for each key/value pair supplied.
+     * <p/>
+     * If a {@link Map} is not supplied then an exception is thrown.
+     * <p /> 
+     * This method is effectively calls {@link #property(VertexProperty.Cardinality, Object, Object, Object...)}
+     * as {@code property(null, key, value, keyValues}.
+     *
+     * @param value     the value for the property
+     * @return the traversal with the last step modified to add a property
+     * @see <a href="http://tinkerpop.apache.org/docs/${project.version}/reference/#addproperty-step" target="_blank">AddProperty Step</a>
+     * @since 3.0.0-incubating
+     */
+    public default GraphTraversal<S, E> property(final Object value) {
+        if (value instanceof Map) { //Handle the property(Cardinality, Map) signature
+            final Map<Object, Object> map = (Map)value;
+            for (Map.Entry<Object, Object> entry : map.entrySet()) {
+                property(null, entry.getKey(), entry.getValue());
+            }
+            return this;
+        } else {
+            throw new IllegalArgumentException("property(object) must have a Map argument");
+        }
     }
-
```

Comment:
nit:  looks like some unintended changes here/below here in `GraphTraversal.java`

Reply:
Yeah not sure how that happened, reverted